code/modules/spells/spell_types/hivemind.dm

```diff
@@ -0,0 +1,624 @@
+		if(!user.getBruteLoss() && !user.getFireLoss() && !user.getCloneLoss()) //If we don't have any of these, stop looping
+			to_chat(user, "<span class='warning'>We finish our healing</span>")
+			break
+		iterations++
+	user.setBrainLoss(0)
+
+
+/obj/effect/proc_holder/spell/target_hive/hive_control
+	name = "Mind Control"
+	desc = "We assume direct control of one of our vessels, leaving our current body for up to ten seconds, although a larger hive may be able to sustain it for up to two minutes. Powers can be used via our vessel, although if it dies, the entire hivemind will come down with it."
+	charge_max = 600
+	action_icon_state = "force"
+	active  = FALSE
+	var/mob/living/carbon/human/original_body //The original hivemind host
+	var/mob/living/carbon/human/vessel
+	var/mob/living/backseat //Storage for the mind controlled vessel
+	var/power = 100
+	var/time_initialized = 0
+
+/obj/effect/proc_holder/spell/target_hive/hive_control/proc/release_control() //If the spell is active, force everybody into their original bodies if they exist, ghost them otherwise, delete the backseat
+	if(!active)
+		return
+	active = FALSE
+	charge_counter = max((0.5-(world.time-time_initialized)/power)*charge_max, 0) //Partially refund the power based on how long it was used, up to a max of half the charge time
+
+	if(!QDELETED(vessel))
+		if(vessel.mind)
+			if(QDELETED(original_body))
+				vessel.ghostize(0)
+			else
+				vessel.mind.transfer_to(original_body, 1)
+
+	if(!QDELETED(backseat) && backseat.mind)
+		backseat.remove_trait(TRAIT_MUTE)
+		if(QDELETED(vessel))
+			backseat.ghostize(0)
+		else
+			backseat.mind.transfer_to(vessel,1)
+
+	message_admins("[ADMIN_LOOKUPFLW(vessel)] is no longer being controlled by [ADMIN_LOOKUPFLW(original_body)] (Hivemind Host).")
+	log_game("[key_name(vessel)] was released from Mind Control by [key_name(original_body)].")
+
+	QDEL_NULL(backseat)
+
+/obj/effect/proc_holder/spell/target_hive/hive_control/on_lose(mob/user)
+	release_control()
+
+/obj/effect/proc_holder/spell/target_hive/hive_control/cast(list/targets, mob/living/user = usr)
+	if(!active)
+		vessel = targets[1]
+		var/datum/antagonist/hivemind/hive = user.mind.has_antag_datum(/datum/antagonist/hivemind)
+		if(!hive)
+			to_chat(user, "<span class='notice'>This is a bug. Error:HIVE1</span>")
+			return
+		switch(hive.hive_size)
+			if(10 to 14)
+				power = 100
+				charge_max = 600
+			if(15 to 19)
+				power = 300
+				charge_max = 900
+			if(20 to 24)
+				power = 600
+				charge_max = 1200
+			else
+				power = 1200
+				charge_max = 1200
+		for(var/datum/antagonist/hivemind/H in GLOB.antagonists)
+			if(H.owner == user.mind)
+				continue
+			if(H.hivemembers.Find(vessel))
+				to_chat(user, "<span class='danger'>We have detected a foreign presence within this mind, it would be unwise to merge so intimately with it.</span>")
+				revert_cast()
+				return
+		original_body = user
+		vessel = targets[1]
+		backseat = new /mob/living()
```

Comment:
This could use specific subtype. (Shared by split personality stuff ?)

Reply:
The split personality subtype overrides Life() so that if somebody ghosts, the other mob stays forever, and the whole subtype is tied into the owner's trauma. Made a new subtype for it instead.